spec/controllers/users_controller_spec.rb

```diff
@@ -68,15 +78,38 @@
     end
   end
 
-  describe 'DELETE users#destroy' do
-    it 'deletes the user' do
+  describe 'users#destroy' do
+    it 'deletes the current_user account' do
       expect(response).to have_http_status(:ok)
       expect { delete :destroy, params: { id: user.id } }.to change(User, :count).by(-1)
     end
 
-    it 'does not delete any user if the user id is invalid' do
+    it 'returns status code forbidden if the user tries to delete another user' do
+      new_user = create(:user)
+      expect { delete :destroy, params: { id: new_user.id } }.not_to change(User, :count)
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it 'returns status code forbidden if the user id is invalid' do
       expect { delete :destroy, params: { id: 'invalid-id' } }.not_to change(User, :count)
-      expect(response).to have_http_status(:not_found)
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    context 'current_user with ManageUsers permission deletes another user' do
+      before do
+        manage_users_role_permission
```

Comment:
Is this line required?

Reply:
Yes, it's needed to instantiate the object.
I would use let! instead but RSpec is complaining that the object is not being directly referenced - maybe we should remove that rule?

Reply:
Yeah I agree that rule is annoying

Reply:
Yeah I would remove it and use `let!`